src/components/lib/cos_kernel_api.c

```diff
@@ -200,7 +201,7 @@ __capid_captbl_check_expand(struct cos_compinfo *ci)
 
 	printd("__capid_captbl_check_expand->pre-captblactivate (%d)\n", CAPTBL_OP_CAPTBLACTIVATE);
 	/* captbl internal node allocated with the resource provider's captbls */
-	if (call_cap_op(meta->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, captblcap, meta->pgtbl_cap, kmem, 1)) {
+	if (call_cap_op(meta->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, captblcap, meta->local_mi.pgtbl_cap, kmem, 1)) {
```

Comment:
I guess, like Gabe mentioned before, `OP_CONS` which is called next must take in an additional parameter that indicates which `pgtbl/captbl` has this `captblcap`.